kext/FileNVRAM/FileNVRAM.cpp

```diff
@@ -217,7 +217,7 @@ void FileNVRAM::copyUnserialzedData(const char* prefix, OSDictionary* dict)
             
             if(prefix)
             {
-                size_t size = strlen(prefix) + sizeof(NVRAM_SEPERATOR) + strlen(key->getCStringNoCopy());
+                size_t size = strlen(prefix) + sizeof(NVRAM_SEPERATOR) + strlen(key->getCStringNoCopy()) +1;
```

Comment:
Sounds good.

For reference:
- The kext was never meant to be used standalone. The reason for reading in the plist two-fold:
(1) As a backup in case if the bootloder (chameleon, or potentially clover if bootloader support is added) has an issue. The NVRAM contents really do need to be loaded in the bootloader, as certain processes *do* read the nvram the file-ystem is mounted.
(2) I often run the kext on real macs during development and to allow me to trap internal behavior of programs.

In any case,
Let's get this pull request merged in before you make the iobsd changes. I'll go though the latest changes and see if there's anything else.